src/user/user.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { Role } from 'src/shared/enums/user-role.enum';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { User, UserDocument } from './schemas/user.schema';
+
+@Injectable()
+export class UserService {
+  constructor(@InjectModel(User.name) private userModel: Model<UserDocument>) {}
+  
+  async create(createUserDto: CreateUserDto) {
+    const user = new this.userModel(createUserDto);
+    
+    return user.save();
+  }
+
+  async findAll() {
+    return this.userModel.find().exec();
+  }
+
+  async findOne(options: Partial<Omit<User, 'password'>>) {
+    return this.userModel.findOne(options).exec();
```

Comment:
Wouldn't it be easier to find it just by `id` 😄 ? Sounds enough `unique` for finding it

Reply:
Besides, it would allows deleting the `findById` class method or replacing this one with that one

Reply:
It’s is just a generic abstraction to query over users -it’s not meant to be used directly, but by functions with more specific queries. You don’t always have id of the user, or want to get all users with selected properties.